cpp/src/join/hash_join.cu

```diff
@@ -64,28 +78,25 @@ std::unique_ptr<multimap_type, std::function<void(multimap_type*)>> build_join_h
   size_type const build_table_num_rows{build_device_table->num_rows()};
   std::size_t const hash_table_size = compute_hash_table_size(build_table_num_rows);
 
-  auto hash_table = multimap_type::create(hash_table_size,
-                                          stream,
-                                          true,
-                                          multimap_type::hasher(),
-                                          multimap_type::key_equal(),
-                                          multimap_type::allocator_type());
+  auto hash_table = std::make_unique<multimap_type>(
+    hash_table_size, std::numeric_limits<hash_value_type>::max(), JoinNoneValue);
```

Comment:
I don't think the `unique_ptr` is necessary here. You can just construct the `static_map` as normal.